x-pack/plugins/observability_solution/observability_shared/public/components/feature_feedback_button/feature_feedback_button.test.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { type FormConfig, getSurveyFeedbackURL, NodeType } from './feature_feedback_button';
+
+describe('getSurveyFeedbackURL', () => {
+  const formUrl = 'https://ela.st/foo';
+
+  it('should return the correct URL without any parameters', () => {
+    const expectedUrl = formUrl;
+    const actualUrl = getSurveyFeedbackURL({ formUrl });
+    expect(actualUrl).toBe(`${expectedUrl}?entry.573002982=Self-Managed+%28you+manage%29`);
+  });
+
+  it('should append kibana version parameter correctly', () => {
```

Comment:
Thank you for adding the tests! I am trying to explain the logic (also added this comment: https://github.com/elastic/kibana/pull/179096/files#r1534173725)
`isCloudEnv` and `isServerlessEnv` should be defined and equal `false` not `undefined` here (or at least one should be defined). 
So the cases we have are:
`isCloudEnv` and `isServerlessEnv` are undefined - do not prefill
`isCloudEnv` and `isServerlessEnv` are `false` we have `Self-Managed`
`isCloudEnv` and `isServerlessEnv`  are both `true` or `isCloudEnv` is` false` (I think in this case they both should be true in theory) then it's `Serverless`
`isCloudEnv` is `true` and `isServerlessEnv` is `false` or not defined then we have `Elastic Cloud`